src/udiskslinuxfilesystem.c

```diff
@@ -183,6 +186,13 @@ udisks_linux_filesystem_update (UDisksLinuxFilesystem  *filesystem,
   dev = udisks_linux_block_object_get_device_file (object);
   type = g_udev_device_get_property (device->udev_device, "ID_FS_TYPE");
   size = 0;
+
+  //skip get_info when drive is suspended
+  if (!get_pm_state(device, &error, &count))
```

Comment:
This function is ATA-specific and is being called for any kind of device at this place. In many cases it would fail, skipping the following code, which is wrong. The only case the following code should be skipped is when the ATA command succeeds and returns a value indicating the drive is sleeping.